core/src/main/java/org/radargun/stages/cache/background/AbstractLogLogic.java

```diff
@@ -219,28 +230,50 @@ protected void afterRollback() {
       delayedRemoves.clear();
    }
 
-   protected void afterCommit() {
+   protected boolean afterCommit() {
       boolean inTransaction = false;
       try {
+         Map<Long, Integer> delayedRemoveAttemptsMap = new HashMap<>();
          while (!stressor.isTerminated()) {
             try {
-               if (inTransaction) {
+               if (inTransaction || ongoingTx != null) {
                   try {
                      ongoingTx.rollback();
                   } catch (Exception e) {
-                     log.error("Failed to rollback ongoing transaction", e);
+                     log.error("Failed to rollback ongoing transaction.", e);
                   }
                }
                startTransaction();
                inTransaction = true;
                for (DelayedRemove delayedRemove : delayedRemoves.values()) {
-                  checkedRemoveValue(delayedRemove.keyId, delayedRemove.oldValue);
+                  try {
```

Comment:
Thinking about it now, such key differentiation really brings little added value, fixed.